tket2/src/portmatching/pattern.rs

```diff
@@ -42,17 +43,17 @@ impl CircuitPattern {
             let op = cmd.optype().clone();
             pattern.require(cmd.node().into(), op.try_into().unwrap());
             for in_offset in 0..cmd.input_count() {
-                let in_offset = Port::new_incoming(in_offset);
-                let edge_prop =
-                    PEdge::try_from_port(cmd.node(), in_offset, circuit).expect("Invalid HUGR");
+                let in_offset: IncomingPort = in_offset.into();
+                let edge_prop = PEdge::try_from_port(cmd.node(), in_offset.into(), circuit)
```

Comment:
I think you can do this more directly (avoiding `in_offset`) via `Port::new_incoming`

Reply:
We use `in_offset` below for `linked_outputs`, so we need the Incoming port.